tansanchian iP

tansanchian opened the pull request:
## HandsomePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

HandsomePro frees your mind of having to remember things you need to do. It's,
+ text-based
+ easy to learn
+ ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your tasks.
4. let it manage your task for you :wink:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

__ __

If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` methd:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

tansanchian committed:
Rename, Greet, Exit

tansanchian committed:
Echo

tansanchian committed:
Add, List

tansanchian committed:
Mark as Done

tansanchian committed:
ToDos, Events, Deadlines

tansanchian committed:
Automated Text UI Testing

tansanchian committed:
Handle Errors

tansanchian committed:
Delete

tansanchian committed:
Save the tasks in hard disk

When the chatbot starts up for the first time, a directory with a text
file, handsome.txt will be created and tasks will be saved automatically
in the file whenever the task list changes. Future starting up of the
bot will load the data from the file as task list.

tansanchian committed:
Merge branch 'branch-Level-7'

tansanchian committed:
Fix bugs of data format while writing in handsome.txt

The data was always recorded as ToDo Task regardless of which type of
task is given and has now been fixed

tansanchian committed:
Format the dates and times

The chatbot can now print date with a format of Day, Date, Time. For
example, 2024-05-24 0600 will be formatted to be Fri, 24 May 2024,
06:00 AM

tansanchian committed:
Merge branch 'branch-Level-8'

tansanchian committed:
Make the code more OOP

Refactor the code to extract out closely related code as classes.

tansanchian committed:
Merge branch 'branch-A-MoreOOP'

tansanchian committed:
Divide classes into packages

Organize the classes into suitable java packages.

tansanchian committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2425S1/ip

tansanchian committed:
Use Gradle to automate project builds

Use Gradle to automate some of the build tasks of the project, including
running the bot using Gradle and also check style.

tansanchian committed:
Add JUnit tests

Add JUnit tests to test the behavior of the code. In particular, the
task and command classes are tested reasonably well.

tansanchian committed:
Package the App as a JAR file

Package the app as an executable JAR file so that it can be distributed
easily.

tansanchian committed:
Add JavaDoc comments

Add JavaDoc comments to all classes and methods

tansanchian committed:
Merge branch 'branch-A-JavaDoc'

tansanchian committed:
Tweak the code to comply with a coding standard

tansanchian committed:
Implement find functionality

Give users a way to find a task by searching for a keyword in the task
description.

tansanchian committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/handsome/command/MarkCommand.java
#	src/main/java/handsome/command/TodoCommand.java
#	src/main/java/handsome/command/UnmarkCommand.java

tansanchian committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/handsome/task/Task.java

tansanchian committed:
Add a GUI to the chatbot.

Used the JavaFX technology to implement the GUI. Remove cetain parts of
the CLI while some lines are still printed in the terminal.

tansanchian committed:
Merge branch 'branch-Level-10'

M-kew commented on src/main/java/handsome/task/Task.java:
```diff
@@ -0,0 +1,83 @@
+        String[] dateAndTime = date.split(" ");
+        LocalDate d = LocalDate.parse(dateAndTime[0]);
+        LocalTime t = LocalTime.parse(dateAndTime[1], DateTimeFormatter.ofPattern("HHmm"));
+        return d.getDayOfWeek().getDisplayName(TextStyle.SHORT, Locale.ENGLISH) + ", " + d.getDayOfMonth() + " "
```
Maybe you could write a comment explaining what this code does as it would be hard to understand at first glance :)

M-kew commented on src/main/java/handsome/task/Event.java:
```diff
@@ -0,0 +1,61 @@
+    private final String endTime;
+
+    /**
+     * Constructs an Event object with the specified content, status, start time, and end time.
```
Very detailed javadocs! 👍

M-kew commented on src/main/java/handsome/task/Task.java:
```diff
@@ -0,0 +1,83 @@
+     * @param keyword The keyword to search for within the task's content.
+     * @return true if the task's content contains the keyword, false otherwise.
+     */
+    public boolean contains(String keyword) {
```
the naming of the boolean method should have prefix (e.g. isFound, hasData).

M-kew reviewed (commented):
LGTM. Just some minor changes to be made. Good work on following the coding standard! :)

M-kew commented on src/main/java/handsome/command/Command.java:
```diff
@@ -0,0 +1,61 @@
+     * @throws HandsomeException If there is an error during the command execution.
+     * @throws IOException       If an I/O error occurs while interacting with the storage.
+     */
+    public abstract String execute(TaskList tasks, Storage storage, Ui ui) throws HandsomeException, IOException;
```
Good use of plural form to represent collection of objects! :)

tansanchian committed:
Change the main class name of build.gradle

Change the main class name of build.gradle to generate a new JAR file
using Gradle.

tansanchian committed:
Use assertions to document preconditions and postconditions

I've added assert statements to check for non-null parameters in methods
across different classes (TaskList, Storage, Command, etc.). This
ensures that no method operates on null objects, preventing potential
NullPointerExceptions.

This has improved the code quality, robustness, and maintainability. It
ensures that the code behaves as expected under all circumstances.

tansanchian committed:
Improve code quality

Critically examines the code and refactor to improve the code quality
where necessary.

tansanchian committed:
Merge pull request #1 from tansanchian/branch-A-Assertions

Use assertions to document preconditions and postconditions

tansanchian committed:
Merge branch 'master' of https://github.com/tansanchian/ip

* 'master' of https://github.com/tansanchian/ip:
  Use assertions to document preconditions and postconditions

tansanchian committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/MainWindow.java
#	src/main/java/handsome/TaskList.java

tansanchian committed:
Merge pull request #2 from tansanchian/branch-A-CodeQuality

Improve code quality

tansanchian committed:
Merge branch 'master' of https://github.com/tansanchian/ip

* 'master' of https://github.com/tansanchian/ip:
  Improve code quality

tansanchian committed:
Add archive and unarchive methods to the code base

Users are now able to archive the task list so that those archived tasks
will not shown in the list. Unarchive will then bring back all the
existing tasks.

With this, users can start over with a clean state and retrieve those
tasks when they need.

tansanchian committed:
Add archive and unarchive methods to the code base

Users are now able to archive the task list so that those archived tasks
will not shown in the list. Unarchive will then bring back all the
existing tasks.

With this, users can start over with a clean state and retrieve those
tasks when they need.

tansanchian committed:
Improve the GUI to make it more polished

tansanchian committed:
Add a user guide to the project in README.md

tansanchian committed:
Update some typos in the user guide

tansanchian committed:
Update and polish the user guide

tansanchian committed:
Update README.md
